src/components/LocaleToggle/LocaleToggle.tsx

```diff
@@ -8,24 +8,37 @@ const LocaleToggle: FC = () => {
     const [locale, setLocale] = useState(Locale.EN);
     const { i18n } = useTranslation();
 
-    const handleChange = (_: React.MouseEvent<HTMLElement>, nextLocale: Locale) => {
-        const changeLanguage = async () => {
-            await i18n.changeLanguage(nextLocale).then(() =>
-                setLocale(nextLocale)
-            )
+    const changeLanguage = async (nextLocale: Locale) => {
+        await i18n.changeLanguage(nextLocale).then(() => {
+            setLocale(nextLocale);
+            localStorage.setItem('locale', nextLocale);
+        });
+    }
+
+    useEffect(() => {
+        const savedLocale = localStorage.getItem('locale');
+        if (savedLocale) {
+            changeLanguage(savedLocale as Locale).catch((e: Error) => {
+                console.error(e)
+            })
         }
-        changeLanguage().catch((e: Error) => {
+    }, []);
```

Comment:
```suggestion
    }, [changeLanguage]);
```

Need to add the dependency to the hook here

Reply:
Thanks, I had to add a useCallback hook to wrap the changeLanguage function too